src/main/java/com/mongodb/spark/sql/connector/schema/BsonDocumentToRowConverter.java

```diff
@@ -250,6 +251,11 @@ private boolean convertToBoolean(
     return bsonValue.asBoolean().getValue();
   }
 
+  private Date convertToDate(
+      final String fieldName, final DataType dataType, final BsonValue bsonValue) {
+    return new Date(convertToLong(fieldName, dataType, bsonValue));
```

Comment:
Do we have to use `java.sql.Date` because not all versions of Spark we support always support `java.time.LocalDate`?

---

Found the answer, it's "yes":

- Spark 3.1.2 documents only `java.sql.Date` as the Java type for `DataTypes.DateType` https://spark.apache.org/docs/3.1.2/sql-ref-datatypes.html
- Spark 3.5.1 documents also `java.time.LocalDate` as the Java type for `DataTypes.DateType` https://spark.apache.org/docs/3.5.1/sql-ref-datatypes.html